### PR TITLE
Better error messages in interpolate_for_nondim_position

### PR DESCRIPTION
**Pull request by Hallberg-NOAA**
Cleaned up the error messages when there are fatal problems in `interpolate_for_nondim_position()` in the MOM_neutral_diffusion module to explicitly give an indications of all the problems.  These had previously triggered fatal errors (or should have), but had less explicit or incorrect error messages in some cases, so all solutions and behavior are identical in any cases that worked previously.

### CODE REVIEW

**Review by marshallward — CHANGES_REQUESTED** [review]
I believe that the conditionals can be simplified due to the boundedness of `interpolate_for_nondim_position`, and the error message can be consolidated into a single block.

While this does not need to be addressed now, we should explore if this function might be written without branching, and without `MOM_error` calls.  Both coule be making vectorization and GPU migration more difficult.

File: src/tracer/MOM_neutral_diffusion.F90 [hunk 1]
```diff
@@ -1586,10 +1577,22 @@ real function interpolate_for_nondim_position(dRhoNeg, Pneg, dRhoPos, Ppos)
...
     interpolate_for_nondim_position = 0.5
   endif
-  if ( interpolate_for_nondim_position < 0. ) &
-    call MOM_error(FATAL, 'interpolate_for_nondim_position: Houston, we have a problem! Pint < Pneg')
-  if ( interpolate_for_nondim_position > 1. ) &
-    call MOM_error(FATAL, 'interpolate_for_nondim_position: Houston, we have a problem! Pint > Ppos')
+
+  ! Error handling for problematic cases.  It is expected that this should never occur.
+  if ( (Ppos < Pneg) .or. (dRhoNeg > dRhoPos) .or. &
+       (interpolate_for_nondim_position < 0.) .or. (interpolate_for_nondim_position > 1.) ) then
```
[the remark this answers is not in the document]
Comment (Hallberg-NOAA) [reply src/tracer/MOM_neutral_diffusion.F90:1583]: These are all good questions.  Probably the right person to answer them would be @ashao, who wrote the original version of this code.

  I agree with your analysis of how the logic could be simplified.  Everything in this commit is retaining the original logic, but just making the messages more explicit.  I would be happy to modify this as suggested, but before doing so, I would prefer to hear what @ashao thought about this.

**Comment by Hallberg-NOAA** [comment]
This PR has been revised as suggested.

**Review by marshallward — APPROVED** [review]

**Comment by marshallward** [comment]
Gaea regression: https://gitlab.gfdl.noaa.gov/ogrp/mom6ci/MOM6/-/pipelines/24069 :heavy_check_mark:

**Pull request merged**